### PR TITLE
Fix Custom image selector

### LINKED ISSUE #5558: Custom image selector: Do not reload everything after marking a picture as not for upload
How to reproduce the issue:

0. In the custom image selector, mark your first 100 pictures as not for upload, then disable "Show already actioned pictures".
1. (screencast below starts at this step) In the custom image selector, tap one picture.
2. Tap "Mark as not for upload"

Expected: Only the selected picture diapsears from the screen. The remaining pictures move up a bit, one additional picture gets shown at the bottom.
 
Actually: All pictures disappear from the screen, the activity gets reloaded fully which takes a lot of time (about 20 seconds in the screencast below).

https://github.com/commons-app/apps-android-commons/assets/99590/3c17a3ef-c13a-4852-8c73-21e63c65c921

### PR DESCRIPTION
**Pull request by shashankiitbhu**
**Description (required)**

Fixes #5558 

What changes did you make and why?

**Tests performed (required)**

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

**Screenshots (for UI changes only)**

https://github.com/commons-app/apps-android-commons/assets/126143257/14863ad7-5d4e-4abc-9119-f7c9d019b18f

### CODE REVIEW

File: app/src/main/res/values-yue-hant/error.xml
Comment (neeldoshii) [inline app/src/main/res/values-yue-hant/error.xml:1]: Why is this file removed? 

 ```
 AGPBI: {"kind":"error","text":"Invalid resource directory name",
Execution failed for task ':app:mergeBetaDebugResources'.
> /project-directory/apps-android-commons/app/src/main/res/values-yue-hant: Error: Invalid resource directory name
```
I know there is a issue of Gerrit, but removing this file or editing the name of the file is temporary fix as this file will get auto generated by translate wiki. 

[Originally Comment by @nicolas-raoul ](https://github.com/commons-app/apps-android-commons/issues/5571#issuecomment-1971287512)

**Review by neeldoshii — CHANGES_REQUESTED** [review]

File: app/src/main/res/values-yue-hant/error.xml
Comment (shashankiitbhu) [reply app/src/main/res/values-yue-hant/error.xml:1]: I am aware of that issue, I have removed this as a temporary fix (mentioned in the original issue there) so that it can be built here and tests can be performed (as that issue is yet to be resolved)

Comment (neeldoshii) [reply app/src/main/res/values-yue-hant/error.xml:1]: > I am aware of that issue, I have removed this as a temporary fix (mentioned in the original issue there) so that it can be built here and tests can be performed (as that issue is yet to be resolved)

Don't add here file removed in the commit do it locally but don't push. Test cases will still be performed without deleting the file.

Comment (neeldoshii) [reply app/src/main/res/values-yue-hant/error.xml:1]: > I am aware of that issue, I have removed this as a temporary fix (mentioned in the original issue there) so that it can be built here and tests can be performed (as that issue is yet to be resolved)

Ok gotcha.
```
Execution failed for task ':app:mergeBetaDebugResources'.
> /home/runner/work/apps-android-commons/apps-android-commons/app/src/main/res/values-yue-hant: Error: Invalid resource directory name
```

**Comment by nicolas-raoul** [comment]
Testing now...

Sometimes after marking many pictures as "not for upload" I see a large number of white squares at the bottom, not sire whether it is intended?
Also I had an instance where the last loaded picture at the bottom was not selectable.

This is a huge improvement, I already feel much more efficient thanks to this change. :-) I continue testing.

File: app/src/main/res/values-yue-hant/error.xml
Comment (nicolas-raoul) [reply app/src/main/res/values-yue-hant/error.xml:1]: Unfortunately the issue is not getting solved...
To not prevent everyone from sending pull requests, I will accept pull requests that fix an issue and at the same time delete the values-yue-hant directory to prevent the build from failing.

**Comment by nicolas-raoul** [comment]
Examples:

File: app/src/main/java/fr/free/nrw/commons/customselector/ui/adapter/ImageAdapter.kt [hunk 1]
```diff
@@ -364,6 +367,47 @@ class ImageAdapter(
         notifyDataSetChanged()
     }
 
+    fun clearSelectedImages(){
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/customselector/ui/adapter/ImageAdapter.kt:370]: Would you mind adding doc to this method and the next?

File: app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageFragment.kt [hunk 2]
```diff
@@ -339,6 +339,13 @@ class ImageFragment : CommonsDaggerSupportFragment(), RefreshUIListener, PassDat
         imageAdapter.refresh(filteredImages, allImages)
     }
 
+    fun removeImage(image : Image){
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageFragment.kt:342]: Would you mind adding doc to this method and the next?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/customselector/ui/adapter/ImageAdapter.kt (hunk 1 above)
Comment (shashankiitbhu) [reply app/src/main/java/fr/free/nrw/commons/customselector/ui/adapter/ImageAdapter.kt:370]: sure

**Comment by shashankiitbhu** [comment]
> Examples:
> 
> 
> 
> 

On My device, these white squires appear for a short time only (until they are loaded, and this happens only for folders with a lot of Images ), also these aren't empty white squares they are also loaded Images but for some of them the thumbnail processing takes a bit more time  (which you can perform for any of those white squares immediately by clicking on any of them), I have tested it and came to a conclusion that all Images are loaded but some of them take some time to load the thumbnails.

Previously loading images was too slow and it had enough to process each thumbnail but now image is loaded much faster so it takes some time to load.

**Comment by nicolas-raoul** [comment]
Yes I also observed that moving the empty squares out of visible screen, and then back into visible screen, makes them load correctly. Waiting does not seem to make them appear.

Another small issue: pictures marked as "not for upload" are removed from the view even if "show already actioned pictures" is enabled:
https://github.com/commons-app/apps-android-commons/assets/99590/c148155b-0862-40ea-98bb-be9dd0bc54fd

Both are not big issues compared to the benefit brought by this PR, so I am happy to file 2 new issues and merge for now, if you agree. 🙂

**Comment by shashankiitbhu** [comment]
@nicolas-raoul Sure , I can work on them separately, and given my understanding of this issue and changes I have made , I think we'll get reliable solutions soon

**Comment by shashankiitbhu** [comment]
> Another small issue: pictures marked as "not for upload" are removed from the view even if "show already actioned pictures" is enabled:

Fixed this one in the latest Commit:


https://github.com/commons-app/apps-android-commons/assets/126143257/02190178-9ac3-4856-a476-40a39246f1c2

**Comment by shashankiitbhu** [comment]
@nicolas-raoul If the Above works can you please review and merge it, thanks

**Review by nicolas-raoul — APPROVED** [review]
A few bugs subsist but hugely helpful fix, thank you! :-)
